test/replication-luatest/gh_7932_recovery_mixed_transactions_test.lua

```diff
@@ -0,0 +1,167 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local fio = require('fio')
+
+local g = t.group('gh_7932')
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new({})
+    cg.server = cg.cluster:build_and_add_server{
+        alias = 'server',
+        box_cfg = {
+            instance_uuid = '1a685e63-24cd-4fc5-9532-bf85f649e0ab',
+        },
+    }
+    cg.cluster:start()
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+-- Check that transactions with both local and global spaces recover fine
+-- with this patch
+g.test_local_and_global_spaces_recover = function(cg)
+    cg.server:exec(function()
+        box.schema.space.create('loc', {is_local = true})
+        box.space.loc:create_index('pk')
+        box.schema.space.create('glob')
+        box.space.glob:create_index('pk')
+        box.begin()
+        for i = 1,3 do
+            box.space.glob:replace{i, box.info.replication[1].id}
+            box.space.loc:replace{i, 0}
+        end
+        box.commit()
+    end)
+
+    t.assert_equals(cg.server:exec(function()
+        return box.space.glob:select()
+    end), {{1, 1}, {2, 1}, {3, 1}})
+    t.assert_equals(cg.server:exec(function()
+        return box.space.loc:select()
+    end), {{1, 0}, {2, 0}, {3, 0}})
+
+    cg.server:stop()
+    local xlog = fio.pathjoin(cg.server.workdir,
+                              '00000000000000000012.xlog')
+    fio.unlink(xlog)
```

Comment:
Why do you need to unlink the xlog here?

Reply:
Can be left.